linux_os/guide/services/sssd/sssd_enable_pam_services/bash/shared.sh

```diff
@@ -6,7 +6,7 @@
 SSSD_CONF="/etc/sssd/sssd.conf"
 SSSD_CONF_DIR="/etc/sssd/conf.d/*.conf"
 
-for f in $( ls $SSSD_CONF $SSSD_CONF_DIR 2> /dev/null ) ; do
+for f in $SSSD_CONF $SSSD_CONF_DIR; do
```

Comment:
This is the wrong approach here. You'd need a glob and an existence check to accomplish this:

```bash
for f in $SSSD_CONF $SSSD_CONF_DIR/*; do
    if [ ! -e "$f" ]; then
        continue
    fi

    ... ## rest of check
done
```

Note that `ls` takes care of the existence check and the directory listing for you.

Reply:
But the glob is already in variable `SSSD_CONF_DIR`
```
 SSSD_CONF_DIR="/etc/sssd/conf.d/*.conf" 
```
so it isn't needed in the `for`, right?

Reply:
But `$SSSD_CONF` could just not exist :-)

Reply:
Also:

```
for x in /tmp /tmp/sodfsdfasdf*; do
                         echo "$x"
                         done
/tmp
/tmp/sodfsdfasdf*
```

Reply:
Edit: Sorry @mildas I misread what you said. yes, no need for the interior glob, but I'm worried shellcheck might complain in general, but maybe its analysis has gotten better understanding of literals/globs. :-)

Reply:
It has gotten smarter since I last used it, nice! :-)

```
 cipherboy   .git  ~  tmp  46  shellcheck file.sh 

In file.sh line 8:
	services_list=$( awk '/^\s*\[/{f=0} /^\s*\[sssd\]/{f=1}f' $f | grep -P '^services[ \t]*=' | uniq )
                                                                  ^-- SC2086: Double quote to prevent globbing and word splitting.

Did you mean: 
	services_list=$( awk '/^\s*\[/{f=0} /^\s*\[sssd\]/{f=1}f' "$f" | grep -P '^services[ \t]*=' | uniq )


In file.sh line 12:
			sed -i "s/$services$/&, pam/" $f
                                                      ^-- SC2086: Double quote to prevent globbing and word splitting.

Did you mean: 
			sed -i "s/$services$/&, pam/" "$f"

For more information:
  https://www.shellcheck.net/wiki/SC2086 -- Double quote to prevent globbing ...
```

Reply:
Ok, I'm going to add the if checks and update the variable